components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultPatientResourceImplTest.java

```diff
@@ -340,23 +338,19 @@ public void deletePatientCatchesXWikiException() throws XWikiException
         }
 
         Assert.assertNotNull("deletePatient did not throw a WebApplicationException as expected "
-            + "when catching an XWikiException", ex);
+            + "when catching an Exception", ex);
         Assert.assertEquals(Status.INTERNAL_SERVER_ERROR.getStatusCode(), ex.getResponse().getStatus());
         verify(this.logger).warn(eq("Failed to delete patient record [{}]: {}"), eq(this.id), anyString());
     }
 
     @Test
     public void deletePatientNormalBehaviour() throws XWikiException
     {
-        XWiki wiki = mock(XWiki.class);
-        XWikiDocument patientXWikiDoc = mock(XWikiDocument.class);
-        doReturn(wiki).when(this.context).getWiki();
-        doReturn(patientXWikiDoc).when(this.patient).getXDocument();
         doReturn(true).when(this.access).hasAccess(Right.DELETE, this.userProfileDocument, this.patientDocument);
 
         Response response = this.patientResource.deletePatient(this.id);
 
-        verify(wiki).deleteDocument(patientXWikiDoc, this.context);
+        verify(this.logger).debug("Deleted patient record [{}]", this.id);
```

Comment:
...also not sure if we want to have this method be declared as `throws XWikiException`, maybe use `WebApplicationException` or catch all exceptions?